### DISCUSSION TITLE
resolution/anti-aliasing in WebGPU vs. WebGL

### DISCUSSION THREAD

**Discussion by ghadeeras** [gpuweb/gpuweb#2088] [Q&A] [answered]
Hello,

Sorry for a rather naïve question. I am new to WebGPU but eager to learn/use it.

I tried migrating a WebGL example I wrote in the past to WebGPU. The example is basically about polygonizing a scalar field isosurface. I managed to have both examples work identically, except I noticed that the quality of the final image is much better in WebGL than WebGPU.

To be more specific, it seems as though the WebGL example uses buffers with higher resolution than the actual canvas, and also does the needed anti-aliasing when down-sampling to the canvas size. Based on that assumption, I tried configuring the 
WebGPU context and the depth buffer to use higher resolution (greater width/height). The overall resolution improved, but I am not sure this is how I am supposed to match the quality of the WebGL example. Besides, even with that, I did not see any anti-aliasing in the down-sampling in WebGPU. The curves are not as smooth.

I tried both examples in Chrome Canary Version 95.0.4630.2. I also attached two sample images of the rendered output to compare.

Is this due to WebGPU still being work in progress, or due to different defaults that I need to be aware of (more configurations to specify)?

WebGL:
![hb-gl](https://user-images.githubusercontent.com/12993823/132138805-ab1b92df-173c-413c-849f-4e43c267df7e.png)

WebGPU:
![hb-gpu](https://user-images.githubusercontent.com/12993823/132138807-60df4206-ff08-422b-8642-aa6ac4f5aed1.png)

**Comment by Kangz** [comment, marked as answer]
There are two potential factors here:

 - WebGL canvas context creation has an `antialias` member that doesn't exist in WebGPU. In WebGPU you need to render antialiased in a separate texture and resolve into the `GPUCanvasContext`'s texture. See how it is done in https://austin-eng.com/webgpu-samples/samples/helloTriangleMSAA but also I think the MSAA attachment should be discarded instead of stored: https://github.com/austinEng/webgpu-samples/issues/140
 - If you're on a high-DPI system you should configure the canvas context with a higher size, and somehow adjust the display ratio of the canvas. I'm not 100% sure how that works though.

**Reply by ghadeeras** [reply]
@Kangz, thanks a lot for your reply! It solved my issue and gave me pointers to more things to explore in WebGPU.
